common/src/main/java/io/druid/common/config/Log4jShutdown.java

```diff
@@ -138,4 +140,55 @@ public boolean isStopped()
   {
     return State.STOPPED.equals(getState());
   }
+
+  private static class StateHolder
+  {
+    @GuardedBy("this")
+    private State current;
+
+    private StateHolder(State initial) {
+      current = initial;
+    }
+
+    private synchronized boolean compareAndSet(State expected, State transition)
+    {
+      if (current == expected) {
+        return transition(transition);
+      }
+      return false;
+    }
+
+    private synchronized boolean waitTransition(State expected, State transition, long timeout)
+    {
+      if (current == expected) {
+        long remaining = timeout;
+        try {
+          long prev = System.currentTimeMillis();
+          while (current != transition && remaining > 0) {
```

Comment:
should this be while(current == expected), wondering if we should return false if some other transition other than the expected happens ?